### DISCUSSION TITLE
Inquiry about Full Response from ESP32 AT SIM Commands

### DISCUSSION THREAD

**Discussion by srikanthpalvai** [espressif/esp-protocols#470] [General]
Dear ESP32 Support Team,

I hope this email finds you well. I am currently working on a project that involves interfacing with ESP32 modules using AT commands.
**https://components.espressif.com/components/espressif/esp_modem**
I have a specific requirement where I need to receive the full response, including the "OK" or "ERROR" status, when executing AT commands.

As an example, when I issue the command to retrieve the IMSI value, I currently receive the following response:
12345678901xxxx

However, for my project, I require the full response, including "OK":

**12345678901xxxx
OK**

I would like to inquire if there is any specific API or configuration setting that would allow me to receive the complete response, including the acknowledgment status.

If such functionality is not currently available, could you please guide me on how I can create a patch file to implement this change? I would like to apply this modification as part of my project build process.

Your guidance and support on this matter would be greatly appreciated.

Thank you for your time and assistance.

Best regards.
Srikanth 
[espressif__esp_modem.patch](https://github.com/espressif/esp-protocols/files/13821429/espressif__esp_modem.patch)

**Comment by david-cermak** [comment]
We've already had a similar request in https://github.com/espressif/esp-protocols/issues/468#issuecomment-1875357122

Thanks for the patch, looks good, but I'd add a new API, just to keep the existing one compatible (as mentioned in the referenced issue)

**Reply by srikanthpalvai** [reply]
Hi David 
Thanks for the quick response, when it will be released, and any estimated date?
If it takes more time, Is there any way I can apply this patch at run time? (idf.py build)

As of now, I have moved this esp_modem component into the project components directory.
But I don't want to do this. i want to apply the patch at runtime, When I do idf.py build this patch should apply in the managed_components/espressif__esp_modem.
Please suggest me 

Thanks in Advance
Srikanth

**Reply by david-cermak** [reply]
Hi Srikanth,

I think we can have a new version sometime next week, this is a simple change.

>  i want to apply the patch at runtime,

It's not possible to apply a patch runtime, but you can use the released version of the component and create your own command. The issue comment referenced above contains all the code and `dce->command()` is a public API.

**Reply by srikanthpalvai** [reply]
Thank you for quick response
